File: codegen/oas-ehr/javascript-closure-angular/API/Client/Contribution.js
goog.provide('API.Client.Contribution');

/**
 * A CONTRIBUTION resource
 * @record
 */
API.Client.Contribution = function() {}

/**
 * @type {!API.Client.HierObjectId}
 * @export
 */
API.Client.Contribution.prototype.uid;

/**
 * @type {!Array<!API.Client.ObjectRef>}
 * @export
 */
API.Client.Contribution.prototype.versions;

/**
 * @type {!API.Client.AuditDetails}
 * @export
 */
API.Client.Contribution.prototype.audit;

File: codegen/swagger-ehr/dart/lib/model/contribution.dart
part of swagger.api;

class Contribution {
  
  HierObjectId uid = null;

  List<ObjectRef> versions = [];

  AuditDetails audit = null;

  Contribution();

  @override
  String toString() {
    return 'Contribution[uid=$uid, versions=$versions, audit=$audit, ]';
  }

  Contribution.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    uid = new HierObjectId.fromJson(json['uid']);
    versions = ObjectRef.listFromJson(json['versions']);
    audit = new AuditDetails.fromJson(json['audit']);
  }

  Map<String, dynamic> toJson() {
    return {
      'uid': uid,
      'versions': versions,
      'audit': audit
     };
  }

  static List<Contribution> listFromJson(List<dynamic> json) {
    return json == null ? new List<Contribution>() : json.map((value) => new Contribution.fromJson(value)).toList();
  }

  static Map<String, Contribution> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, Contribution>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new Contribution.fromJson(value));
    }
    return map;
  }
}
